src/lib/geolocation/client.test.js

```diff
@@ -0,0 +1,43 @@
+/* eslint-disable max-nested-callbacks */
+
+import { expect } from 'chai';
+import {GeolocationClient, GDPR_APPLIES_UNKNOWN, GDPR_APPLIES_NO, GDPR_APPLIES_YES} from "./client";
+
+let geolocationClient;
+
+describe('Geolocation client', () => {
+	beforeEach(() => {
+		window.fetch = null;
+		geolocationClient = new GeolocationClient({domain: 'foo', timeout: 1000});
+	});
+
+	it('gdprCheck returns a promise with a string on 200', (done) => {
+		window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
+			status: 200,
+			text: () => GDPR_APPLIES_NO,
+		}));
+		geolocationClient.gdprCheck()
+			.then(result => {
+				expect(result).to.equal(GDPR_APPLIES_NO);
+				done();
```

Comment:
`done()` shouldn't be necessary with promises
https://jestjs.io/docs/en/asynchronous.html#promises